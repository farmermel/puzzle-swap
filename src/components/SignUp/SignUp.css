form {
  align-items: flex-start;
  display: flex;
  flex-direction: column;
  border: 1px solid #c0c0c0;
  padding: 40px;
  background-color: #fdfdfd;
  width: 50%;
}

form button {
  height: 50px;
  letter-spacing: 2.5px;
  font-size: 2em;
  width: 102%;
}

form h3 {
  font-size: 2em;
  margin: 0 0 10px 0;
}

.form-wrapper {
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 170px 0 250px;
}

.sign-up {
  margin: 0 0 70px 0;
}

#img-submit-label {
  height: 40px;
  background-color: #F6CA4F;
  color: white;
  border-radius: 4px;
  padding: 13px 10px 0px 11px;
  margin: 0 0 16px 0;
}

#img-submit-label:hover {
  background-color: #d1a425;
}

.crop-wrap {
  display: flex;
  flex-direction: column;
}

.crop-label {
  align-self: flex-start;
}

.signup-redirect {
  width: 100%;
  font-size: 1.5em;
  margin: 20px 0 0 0;
}

.signup-redirect span {
  color: #f6ca4f;
}
