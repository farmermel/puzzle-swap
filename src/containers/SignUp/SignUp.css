/*form {
  display: flex;
  flex-direction: column;
  margin: auto;
}*/

.form-wrapper {
  display: flex;
  align-items: center;
  justify-content: center;
}

.sign-up-btn {
  height: 30px;
}
