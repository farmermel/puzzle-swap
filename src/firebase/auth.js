import { auth } from './firebase';

export const doCreateUserWithEmailAndPassword = (email, password) =>
  auth.createUserWithEmailAndPassword(email, password);

export const doSignInWithEmailAndPassword = (email, password) => 
  auth.signInWithEmailAndPassword(email, password);

export const onAuthStateChanged = user => 
  auth.onAuthStateChanged(user);

export const doSignOut = () => 
  auth.signOut();