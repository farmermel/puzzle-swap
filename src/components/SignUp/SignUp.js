import React, { Component } from 'react';
import { auth } from '../../firebase'
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { setLogin } from '../../actions/loginAction';
import './SignUp';

export class SignUp extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      email: '',
      password: '',
      error: null
    }
  }

  handleChange = (e) => {
    this.setState({
      [e.target.name]: e.target.value
    })
  }

  handleSubmit = async (e) => {
    e.preventDefault()
    try {
      const { name, email, password } = this.state;
      const { setLogin } = this.props;
      const authUser = await auth.doCreateUserWithEmailAndPassword(email, password);
      await this.setState({
        name: '',
        email: '',
        password: '',
        error: null
      })
      setLogin(true);
    } catch (e) {
      this.setState({ error: e.message })
      console.log(e)
    }
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}
            className='sign-up'>
        { this.state.error }
        <input type='name' 
               placeholder='name' 
               name='name' 
               onChange={this.handleChange} 
               value={this.state.name} />
        <input type='email' 
               placeholder='email' 
               name='email' 
               onChange={this.handleChange} 
               value={this.state.value} />
        <input type='password' 
               placeholder='password'
               name='password'
               onChange={this.handleChange}
               value={this.state.password} />
        <button className='sign-up-btn'>Sign up</button>
      </form>
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
  setLogin: boolean => dispatch(setLogin(boolean))
});

export default connect(null, mapDispatchToProps)(withRouter(SignUp));