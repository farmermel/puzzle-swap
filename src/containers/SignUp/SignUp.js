import React, { Component } from 'react';
import { auth, db } from '../../firebase'
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { setLogin } from '../../actions/loginAction';
import PropTypes from 'prop-types';
import './SignUp';

export class SignUp extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      email: '',
      password: '',
      error: null
    }
  }

  handleChange = (e) => {
    this.setState({
      [e.target.name]: e.target.value
    })
  }

  handleSubmit = async (e) => {
    e.preventDefault()
    try {
      const { name, email, password } = this.state;
      const { setLogin } = this.props;
      const authUser = await auth.doCreateUserWithEmailAndPassword(email, password);
      await db.doCreateUser(authUser.uid, name, email);
      setLogin(true);
    } catch (error) {
      this.setState({ error: error.message })
      console.log(error)
    }
  }

  render() {
    return (
      <div className='form-wrapper'>
        <form onSubmit={this.handleSubmit}
              className='sign-up'>
          { this.state.error }
          <input type='name' 
                 placeholder='name' 
                 name='name' 
                 onChange={this.handleChange} 
                 value={this.state.name} />
          <input type='email' 
                 placeholder='email' 
                 name='email' 
                 onChange={this.handleChange} 
                 value={this.state.value} />
          <input type='password' 
                 placeholder='password'
                 name='password'
                 onChange={this.handleChange}
                 value={this.state.password} />
          <button className='sign-up-btn'>Sign up</button>
        </form>
      </div>
    )
  }
}

export const mapDispatchToProps = (dispatch) => ({
  setLogin: boolean => dispatch(setLogin(boolean))
});

SignUp.propTypes = {
  setLogin: PropTypes.func.isRequired
}

export default connect(null, mapDispatchToProps)(withRouter(SignUp));