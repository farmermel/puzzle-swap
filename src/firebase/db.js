import { db } from './firebase';

export const doCreateUser = (id, username, email) => 
  db.ref(`users/${id}`).set({
    username,
    email
  });

// export const onceGetUsers = () => 
//   db.ref('users').once('value');

export const getOnce = path =>
  db.ref(path).once('value');

export const getFirebaseKey = childNode => 
  db.ref().child(childNode).push().key;

export const postUpdate = updates => 
  db.ref().update(updates);

export const watchData = watchNode => 
  db.ref(watchNode);

// export const getData = watchNode => 
//   watchData(watchNode).on('value', (snapshot) => {
//     return snapshot.val()
//   })