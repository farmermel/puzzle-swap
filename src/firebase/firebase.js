import * as firebase from 'firebase';
import { firebaseKey,
  fbAuthDomain,
  fbdbUrl,
  fbProjId,
  fbMessagingSenderId } from '../helpers/apiKeys';

const config = {
  apiKey: firebaseKey,
  authDomain: fbAuthDomain,
  databaseURL: fbdbUrl,
  projectId: fbProjId,
  storageBucket: "",
  messagingSenderId: fbMessagingSenderId
};

if (!firebase.apps.length) {
  firebase.initializeApp(config);
}

const auth = firebase.auth();

const db = firebase.database();

export {
  db,
  auth,
};