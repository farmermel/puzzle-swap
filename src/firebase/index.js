import * as auth from './auth';
import * as db from './db';
import * as firebase from './firebase';
import * as store from './store';

export {
  auth,
  db,
  store,
  firebase,
};