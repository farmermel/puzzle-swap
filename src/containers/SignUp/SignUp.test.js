import React from 'react';
import { shallow } from 'enzyme';
import { auth, db } from '../../firebase';
import { SignUp } from './SignUp';

describe('SignUp', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<SignUp />)
  })

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('has default state', () => {
    const expected = {"email": "", "error": null, "name": "", "password": ""};
    expect(wrapper.instance().state).toEqual(expected);
  })

  describe('handleChange', () => {
    it('sets state with event passed in', () => {
      const mockEvent = {
        target: {
          name: 'Mel',
          value: 'is great'
        }
      }
      wrapper.instance().handleChange(mockEvent);
      expect(wrapper.instance().state.Mel).toEqual('is great');
    })
  })

  describe('handleSubmit', () => {
    let mockEvent;
    beforeEach(() => {
      auth.doCreateUserWithEmailAndPassword = jest.fn().mockImplementation(() => {
        return { Casey: 'I am Casey',
        uid: 5}
      })
      db.doCreateUser = jest.fn();
      mockEvent = {
        preventDefault: jest.fn()
      }
    })

    it('prevents default', () => {
      wrapper.instance().handleSubmit(mockEvent);
      expect(mockEvent.preventDefault).toHaveBeenCalled();
    })

    it('calls auth\'s create user method', () => {
      expect(auth.doCreateUserWithEmailAndPassword).not.toHaveBeenCalled();
      wrapper.instance().handleSubmit(mockEvent);
      expect(auth.doCreateUserWithEmailAndPassword).toHaveBeenCalled();
    })

    it('calls db\'s create user method', async () => {
      expect(db.doCreateUser).not.toHaveBeenCalled();
      await wrapper.instance().handleSubmit(mockEvent);
      expect(db.doCreateUser).toHaveBeenCalled();
    })

    it('sets state with error if something fails', async () => {
      auth.doCreateUserWithEmailAndPassword = jest.fn();
      const expected = 'Cannot read property \'uid\' of undefined'
      await wrapper.instance().handleSubmit(mockEvent);
      expect(wrapper.instance().state.error).toEqual(expected);
    })
  })
})