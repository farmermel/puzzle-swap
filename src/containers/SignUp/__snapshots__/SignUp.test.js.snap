// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignUp
    setLogin={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        className="sign-up"
        onSubmit={[Function]}
>
        <input
                name="name"
                onChange={[Function]}
                placeholder="name"
                type="name"
                value=""
        />
        <input
                name="email"
                onChange={[Function]}
                placeholder="email"
                type="email"
                value={undefined}
        />
        <input
                name="password"
                onChange={[Function]}
                placeholder="password"
                type="password"
                value=""
        />
        <button
                className="sign-up-btn"
        >
                Sign up
        </button>
</form>,
      "className": "form-wrapper",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          <input
            name="name"
            onChange={[Function]}
            placeholder="name"
            type="name"
            value=""
/>,
          <input
            name="email"
            onChange={[Function]}
            placeholder="email"
            type="email"
            value={undefined}
/>,
          <input
            name="password"
            onChange={[Function]}
            placeholder="password"
            type="password"
            value=""
/>,
          <button
            className="sign-up-btn"
>
            Sign up
</button>,
        ],
        "className": "sign-up",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "name",
            "onChange": [Function],
            "placeholder": "name",
            "type": "name",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "email",
            "onChange": [Function],
            "placeholder": "email",
            "type": "email",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "password",
            "onChange": [Function],
            "placeholder": "password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign up",
            "className": "sign-up-btn",
          },
          "ref": null,
          "rendered": "Sign up",
          "type": "button",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="sign-up"
          onSubmit={[Function]}
>
          <input
                    name="name"
                    onChange={[Function]}
                    placeholder="name"
                    type="name"
                    value=""
          />
          <input
                    name="email"
                    onChange={[Function]}
                    placeholder="email"
                    type="email"
                    value={undefined}
          />
          <input
                    name="password"
                    onChange={[Function]}
                    placeholder="password"
                    type="password"
                    value=""
          />
          <button
                    className="sign-up-btn"
          >
                    Sign up
          </button>
</form>,
        "className": "form-wrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            null,
            <input
              name="name"
              onChange={[Function]}
              placeholder="name"
              type="name"
              value=""
/>,
            <input
              name="email"
              onChange={[Function]}
              placeholder="email"
              type="email"
              value={undefined}
/>,
            <input
              name="password"
              onChange={[Function]}
              placeholder="password"
              type="password"
              value=""
/>,
            <button
              className="sign-up-btn"
>
              Sign up
</button>,
          ],
          "className": "sign-up",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          null,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "name",
              "onChange": [Function],
              "placeholder": "name",
              "type": "name",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "email",
              "onChange": [Function],
              "placeholder": "email",
              "type": "email",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "password",
              "onChange": [Function],
              "placeholder": "password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign up",
              "className": "sign-up-btn",
            },
            "ref": null,
            "rendered": "Sign up",
            "type": "button",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
